tools/pipelines/templates/upload-json-steps.yml

```diff
@@ -19,24 +19,87 @@ steps:
   clean: true
 
 # Download the api-extractor outputs
-- download: common_definitions
-  artifact: _api-extractor-temp
-- download: common_utils
-  artifact: _api-extractor-temp
-- download: container_definitions
-  artifact: _api-extractor-temp
-- download: core_interfaces
-  artifact: _api-extractor-temp
-- download: driver_definitions
-  artifact: _api-extractor-temp
-- download: protocol_definitions
-  artifact: _api-extractor-temp
-- download: azure
-  artifact: _api-extractor-temp
-- download: client
-  artifact: _api-extractor-temp
-- download: server
-  artifact: _api-extractor-temp
+- task: DownloadPipelineArtifact@2
+  inputs:
+    buildType: specific
+    project: internal
+    pipeline: Build - common-definitions
+    buildVersionToDownload: latestFromBranch
+    branchName: ${{ variables['Build.SourceBranch'] }}
+    artifact: _api-extractor-temp
+    allowPartiallySucceededBuilds: true
```

Comment:
Do we want this enabled? This means we'll download artifacts from potentially broken builds, right?

Reply:
I believe there is `allowFailedBuilds` option which is set to false by default. This option allows us to use builds which completed with warnings. I'm not sure if this is needed on every build, but it will be needed for `core-interfaces` and `driver-definitions` since those pipelines only has archived builds which completed with warnings.